vavr/src/main/java/io/vavr/collection/RedBlackTree.java

```diff
@@ -293,37 +381,47 @@
      * See also <a href="http://n00tc0d3r.blogspot.de/2013/08/implement-iterator-for-binarytree-i-in.html">Implement Iterator for BinaryTree I (In-order)</a>.
      */
     @Override
+    @SuppressWarnings({"unchecked", "rawtypes"})
     default Iterator<T> iterator() {
         if (isEmpty()) {
             return Iterator.empty();
         } else {
             final Node<T> that = (Node<T>) this;
             return new AbstractIterator<T>() {
 
-                List<Node<T>> stack = pushLeftChildren(List.empty(), that);
+                int stackCount = 0;
+                Node<T>[] stack = new Node[that.blackHeight * 2];
```

Comment:
We're guaranteed not to go over this size? I have a hard time verifying that. A comment maybe with a description or invariant for this?

Reply:
done